functional_tests.go

```diff
@@ -373,59 +471,70 @@ func testPutObjectWithMetadata() {
 
 	// Save the data
 	objectName := randString(60, rand.NewSource(time.Now().UnixNano()), "")
+	args["objectName"] = objectName
 
 	// Object custom metadata
 	customContentType := "custom/contenttype"
 
 	n, err := c.PutObjectWithMetadata(bucketName, objectName, reader, map[string][]string{
 		"Content-Type": {customContentType},
 	}, nil)
+	args["metadata"] = map[string][]string{
+		"Content-Type": {customContentType},
+	}
+
 	if err != nil {
-		logger().Fatal("Error:", err, bucketName, objectName)
+		failureLog(function, args, startTime, "", "PutObject failed", err).Fatal()
 	}
 
 	if n != int64(sixtyFiveMiB) {
-		logger().Fatalf("Error: number of bytes does not match, want %v, got %v\n", sixtyFiveMiB, n)
+		failureLog(function, args, startTime, "", "Number of bytes returned by PutObject does not match, want "+string(sixtyFiveMiB)+" got "+string(n), err).Fatal()
 	}
 
 	// Read the data back
 	r, err := c.GetObject(bucketName, objectName)
 	if err != nil {
-		logger().Fatal("Error:", err, bucketName, objectName)
+		failureLog(function, args, startTime, "", "GetObject failed", err).Fatal()
 	}
 
 	st, err := r.Stat()
 	if err != nil {
-		logger().Fatal("Error:", err, bucketName, objectName)
+		failureLog(function, args, startTime, "", "Stat failed", err).Fatal()
 	}
 	if st.Size != int64(sixtyFiveMiB) {
-		logger().Fatalf("Error: number of bytes in stat does not match, want %v, got %v\n",
-			sixtyFiveMiB, st.Size)
+		failureLog(function, args, startTime, "", "Number of bytes returned by PutObject does not match, want "+string(sixtyFiveMiB)+" got "+string(st.Size), err).Fatal()
```

Comment:
`"Number of bytes returned by PutObject does not match, want ...` => `"Number of bytes returned by PutObject does not match GetObject. Expected ...`